plugins/inputs/aliyuncms/aliyuncms.go

```diff
@@ -0,0 +1,597 @@
+	  names = ["InstanceActiveConnection", "InstanceNewConnection"]
+	
+	  ## Dimension filters for Metric (these are optional).
+	  ## This allows to get additional metric dimension. If dimension is not specified it can be returned or
+	  ## the data can be aggregated - it depends on particular metric, you can find details here: https://help.aliyun.com/document_detail/28619.html?spm=a2c4g.11186623.6.690.1938ad41wg8QSq
+	  ##
+	  ## Note, that by default dimension filter includes the list of discovered objects in scope (if discovery is enabled)
+	  ## Values specified here would be added into the list of discovered objects.
+	  ## You can specify either single dimension:      
+	  #dimensions = '{"instanceId": "p-example"}'
+	
+	  ## Or you can specify several dimensions at once:
+	  #dimensions = '[{"instanceId": "p-example"},{"instanceId": "q-example"}]'
+	
+	  ## Enrichment tags, can be added from discovery (if supported)
+	  ## Notation is <measurement_tag_name>:<JMES query path (https://jmespath.org/tutorial.html)>
+	  ## To figure out which fields are available, consult the Describe<ObjectType> API per project.
+	  ## For example, for SLB: https://api.aliyun.com/#/?product=Slb&version=2014-05-15&api=DescribeLoadBalancers&params={}&tab=MOCK&lang=GO
+	  #tag_query_path = [
+	  #    "address:Address",
+	  #    "name:LoadBalancerName",
+	  #    "cluster_owner:Tags.Tag[?TagKey=='cs.cluster.name'].TagValue | [0]"
+	  #    ]
+	  ## The following tags added by default: regionId (if discovery enabled), userId, instanceId.
+	
+	  ## Allow metrics without discovery data, if discovery is enabled. If set to true, then metric without discovery
+	  ## data would be emitted, otherwise dropped. This cane be of help, in case debugging dimension filters, or partial coverage 
+	  ## of discovery scope vs monitoring scope 
+	  #allow_dps_without_discovery = false
+`
+	inputTitle = "inputs.aliyuncms"
+	inputName  = "aliyuncms"
+)
+
+type logger struct {
+	errHeader    string
+	warnHeader   string
+	infoHeader   string
+	debugHeader  string
+	headerFormat string
+}
+
+var lg = newLogger(inputTitle)
+
+func newLogger(header string) *logger {
+	return &logger{
+		headerFormat: "%s %s",
+		errHeader:    fmt.Sprintf("E! [%s]", header),
+		warnHeader:   fmt.Sprintf("W! [%s]", header),
+		infoHeader:   fmt.Sprintf("I! [%s]", header),
+		debugHeader:  fmt.Sprintf("D! [%s]", header)}
+}
+func (l logger) logE(format string, v ...interface{}) {
+	log.Printf(l.headerFormat, l.errHeader, fmt.Sprintf(format, v...))
+}
+func (l logger) logW(format string, v ...interface{}) {
+	log.Printf(l.headerFormat, l.warnHeader, fmt.Sprintf(format, v...))
+}
+func (l logger) logI(format string, v ...interface{}) {
+
+	log.Printf(l.headerFormat, l.infoHeader, fmt.Sprintf(format, v...))
+}
+func (l logger) logD(format string, v ...interface{}) {
+	log.Printf(l.headerFormat, l.debugHeader, fmt.Sprintf(format, v...))
+}
```

Comment:
instead of a custom logger, add 
```
Logger Telegraf.Logger `toml:"-"`
```
to the main struct and it will be populated with a logger instance you can use. Set it to testutils.Logger{} in tests

Reply:
Good point, thank you. Fixed.